### PR TITLE
Refactoring of api/server for reduce engine usage

### PR DESCRIPTION
**Pull request by LK4D4**
Now API server represented by struct `api/server.Server`. It has pointer to daemon instance. All http handlers are methods of that struct, so you can use daemon inside handlers directly.

### CODE REVIEW

**Comment by LK4D4** [comment]
ping @duglin @tiborvass @calavera @runcom

**Comment by icecrime** [comment]
Just to make sure, it means we're changing why this was done in the first place, as described by this comment:

> // load the daemon in the background so we can immediately start		
> // the http api so that connections don't fail while the daemon		
> // is booting

I'm +1, but just making sure.

**Comment by LK4D4** [comment]
Yeah, it is the same. But I actually have much bigger diff for this PR now. I'll post in half-hour.

**Comment by LK4D4** [comment]
@icecrime Ah, sorry, no. We're not changing anything. It was three parallel goroutines: starting daemon, serving api and main. Actually daemon goroutine wasn't necessary.

**Comment by LK4D4** [comment]
Sunday evening ping
@calavera @runcom @tiborvass @duglin

File: api/server/server.go [hunk 1]
```diff
@@ -1600,7 +1633,14 @@ func createRouter(eng *engine.Engine, logging, enableCors bool, corsHeaders stri
...
 // router each time. This requires first moving ListenAndServe into Server.
 func ServeRequest(eng *engine.Engine, apiversion version.Version, w http.ResponseWriter, req *http.Request) {
-	router := createRouter(eng, false, true, "", "")
+	cfg := &ServerConfig{
+		EnableCors: true,
+		Version:    string(apiversion),
+	}
+	api := New(cfg, eng)
+	daemon, _ := eng.HackGetGlobalVar("httpapi.daemon").(*daemon.Daemon)
+	api.AcceptConnections(daemon)
```
Comment (duglin) [inline api/server/server.go]: Can you explain why we need to AcceptConnections() here when we didn't need it before?  I suspect its correct, just trying to follow the flow/change.

Comment (LK4D4) [inline api/server/server.go]: This method used only in `integration` tests, this is why its so ugly. `AcceptConnections` now setting daemon for already serving api.

**Comment by runcom** [comment]
I love this, expecially the removal of `getDaemon(eng)`

**Comment by estesp** [comment]
Should this be in code review? Seems the "design" follows the refactoring that has been happening all over the codebase re: engine/jobs/etc.  This looks good from a quick review standpoint, and I think a server object with daemon accessible makes a lot of sense.

**Comment by LK4D4** [comment]
I would love to merge it. It produces stealth rebase errors on each merge to master :D

File: integration/runtime_test.go [hunk 2]
```diff
@@ -156,6 +156,8 @@ func spawnGlobalDaemon() {
 	globalEngine = eng
```
Comment (duglin) [inline integration/runtime_test.go]: are these still needed?

Comment (LK4D4) [inline integration/runtime_test.go]: I think no. I'll remove it. Just didn't wanna to touch awful integartion tests :)

**Comment by duglin** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM
One more pair of eyes?

**Comment by estesp** [comment]
LGTM

**Comment by runcom** [comment]
yahi!

**Pull request merged**